src/AbstractOperations/grid_metrics.jl

```diff
@@ -148,27 +148,56 @@ GridMetricOperation(L, metric, grid) = GridMetricOperation{L[1], L[2], L[3]}(met
 ##### Spacings
 #####
 
-function xspacings(grid, LX, LY, LZ)
-    Δx_op = KernelFunctionOperation{LX, LY, LZ}(xspacing, grid, LX(), LY(), LZ())
+function xspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δx_op = KernelFunctionOperation{LX, LY, LZ}(xspacing, grid, ℓx, ℓy, ℓz)
     Δx_field = Field(Δx_op)
     compute!(Δx_field)
     return Δx_field
 end
```

Comment:
```suggestion
    Δx_op = KernelFunctionOperation{LX, LY, LZ}(xspacing, grid, ℓx, ℓy, ℓz)
    return Δx_op
end
```

I think this lazy metric is better. The metric fields can get big